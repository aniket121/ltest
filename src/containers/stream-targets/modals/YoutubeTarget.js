import React, { Component } from 'react';
import DropDownMenu from 'material-ui/DropDownMenu';
import MenuItem from 'material-ui/MenuItem';
import TextField from 'components/TextField';
import Modal from 'components/Modal';
import LoaderButton from 'components/LoaderButton';
import ErrorMessage from 'components/Error';
import styles from './styles.css';

class YoutubeTarget extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      name: '',
      privacyStatus: 'public'
    };
  }

  handleNameChange = event => {
    this.setState({
      name: event.target.value
    });
  };

  handlePrivacyStatusChange = (event, index, privacyStatus) => {
    this.setState({ privacyStatus });
  };

  shouldDisableCreate = () => !this.state.name || this.props.isSaving;

  handleSubmit = async event => {
    event.preventDefault();

    const { name, privacyStatus } = this.state;
    const { streamGroupId } = this.props.match.params;
    const { token } = this.props;

    await this.props.createStreamTarget({
      name,
      token,
      provider: 'youtube',
      privacyStatus,
      streamGroupId
    });
  };

  render() {
    const { name } = this.state;
    const { isModalOpen, modalType, closeModal, isSaving, error } = this.props;

    return (
      <Modal
        title="Create Youtube Target"
        open={isModalOpen && modalType === 'youtube_target'}
        closeModal={closeModal}
      >
        <form onSubmit={this.handleSubmit}>
          <div className={styles.modalContent}>
            <TextField
              name="name"
              label="Destination Target Name"
              hint="My youtube channel"
              type="text"
              value={name}
              onChange={this.handleNameChange}
            />

            <DropDownMenu
              name="privacyStatus"
              value={this.state.privacyStatus}
              onChange={this.handlePrivacyStatusChange}
              style={{ width: 310 }}
              selectedMenuItemStyle={{ color: '#6908A1' }}
              autoWidth={false}
            >
              <MenuItem value="public" primaryText="Public" />
              <MenuItem value="private" primaryText="Private" />
              <MenuItem value="unlisted" primaryText="Unlisted" />
            </DropDownMenu>

            <LoaderButton
              type="submit"
              isLoading={isSaving}
              text="Create"
              loadingText="Creating target..."
              disabled={this.shouldDisableCreate()}
            />

            {error && <ErrorMessage message={error.message} link={error.link} />}
          </div>
        </form>
      </Modal>
    );
  }
}

export default YoutubeTarget;
