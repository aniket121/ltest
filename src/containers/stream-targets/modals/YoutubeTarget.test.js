import React from 'react';
import { shallow } from 'enzyme';
import YoutubeTarget from 'containers/stream-targets/modals/YoutubeTarget';
import ErrorMessage from 'components/Error';

const createProps = props => ({
  ...props
});

describe('<YoutubeTarget />', () => {
  test('renders snapshot', () => {
    const wrapper = shallow(<YoutubeTarget />);
    expect(wrapper).toMatchSnapshot();
  });

  test('shows error message if an error exists', () => {
    const props = createProps({ error: 'Failed to create youtube target' });
    const wrapper = shallow(<YoutubeTarget {...props} />);
    expect(wrapper.find(ErrorMessage)).toHaveLength(1);
  })

  test('doesnt show error message if no error exists', () => {
    const props = createProps();
    const wrapper = shallow(<YoutubeTarget {...props} />);
    expect(wrapper.find(ErrorMessage)).toHaveLength(0);
  })
});
