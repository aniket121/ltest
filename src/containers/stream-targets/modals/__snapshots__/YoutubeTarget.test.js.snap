exports[`<YoutubeTarget /> renders snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Modal
    closeModal={undefined}
    open={undefined}
    title="Create Youtube Target">
    <form
        onSubmit={[Function]}>
        <div
            className={undefined}>
            <LivepinTextField
                hint="My youtube channel"
                label="Destination Target Name"
                name="name"
                onChange={[Function]}
                type="text"
                value="" />
            <DropDownMenu
                anchorOrigin={
                    Object {
                        "horizontal": "left",
                        "vertical": "top",
                      }
                }
                animated={true}
                autoWidth={false}
                disabled={false}
                iconButton={<NavigationArrowDropDown />}
                maxHeight={500}
                multiple={false}
                name="privacyStatus"
                onChange={[Function]}
                openImmediately={false}
                selectedMenuItemStyle={
                    Object {
                        "color": "#6908A1",
                      }
                }
                style={
                    Object {
                        "width": 310,
                      }
                }
                value="public">
                <MenuItem
                    anchorOrigin={
                        Object {
                            "horizontal": "right",
                            "vertical": "top",
                          }
                    }
                    checked={false}
                    desktop={false}
                    disabled={false}
                    focusState="none"
                    insetChildren={false}
                    primaryText="Public"
                    value="public" />
                <MenuItem
                    anchorOrigin={
                        Object {
                            "horizontal": "right",
                            "vertical": "top",
                          }
                    }
                    checked={false}
                    desktop={false}
                    disabled={false}
                    focusState="none"
                    insetChildren={false}
                    primaryText="Private"
                    value="private" />
                <MenuItem
                    anchorOrigin={
                        Object {
                            "horizontal": "right",
                            "vertical": "top",
                          }
                    }
                    checked={false}
                    desktop={false}
                    disabled={false}
                    focusState="none"
                    insetChildren={false}
                    primaryText="Unlisted"
                    value="unlisted" />
            </DropDownMenu>
            <LoaderButton
                disabled={true}
                isLoading={undefined}
                loadingText="Creating target..."
                text="Create"
                type="submit" />
        </div>
    </form>
</Modal>,
  "nodes": Array [
    <Modal
      closeModal={undefined}
      open={undefined}
      title="Create Youtube Target">
      <form
            onSubmit={[Function]}>
            <div
                  className={undefined}>
                  <LivepinTextField
                        hint="My youtube channel"
                        label="Destination Target Name"
                        name="name"
                        onChange={[Function]}
                        type="text"
                        value="" />
                  <DropDownMenu
                        anchorOrigin={
                              Object {
                                    "horizontal": "left",
                                    "vertical": "top",
                                  }
                        }
                        animated={true}
                        autoWidth={false}
                        disabled={false}
                        iconButton={<NavigationArrowDropDown />}
                        maxHeight={500}
                        multiple={false}
                        name="privacyStatus"
                        onChange={[Function]}
                        openImmediately={false}
                        selectedMenuItemStyle={
                              Object {
                                    "color": "#6908A1",
                                  }
                        }
                        style={
                              Object {
                                    "width": 310,
                                  }
                        }
                        value="public">
                        <MenuItem
                              anchorOrigin={
                                    Object {
                                          "horizontal": "right",
                                          "vertical": "top",
                                        }
                              }
                              checked={false}
                              desktop={false}
                              disabled={false}
                              focusState="none"
                              insetChildren={false}
                              primaryText="Public"
                              value="public" />
                        <MenuItem
                              anchorOrigin={
                                    Object {
                                          "horizontal": "right",
                                          "vertical": "top",
                                        }
                              }
                              checked={false}
                              desktop={false}
                              disabled={false}
                              focusState="none"
                              insetChildren={false}
                              primaryText="Private"
                              value="private" />
                        <MenuItem
                              anchorOrigin={
                                    Object {
                                          "horizontal": "right",
                                          "vertical": "top",
                                        }
                              }
                              checked={false}
                              desktop={false}
                              disabled={false}
                              focusState="none"
                              insetChildren={false}
                              primaryText="Unlisted"
                              value="unlisted" />
                  </DropDownMenu>
                  <LoaderButton
                        disabled={true}
                        isLoading={undefined}
                        loadingText="Creating target..."
                        text="Create"
                        type="submit" />
            </div>
      </form>
</Modal>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <YoutubeTarget />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": YoutubeTarget {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleNameChange": [Function],
        "handlePrivacyStatusChange": [Function],
        "handleSubmit": [Function],
        "props": Object {},
        "refs": Object {},
        "shouldDisableCreate": [Function],
        "state": Object {
          "isLoading": false,
          "name": "",
          "privacyStatus": "public",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Modal
          closeModal={undefined}
          open={undefined}
          title="Create Youtube Target">
          <form
                    onSubmit={[Function]}>
                    <div
                              className={undefined}>
                              <LivepinTextField
                                        hint="My youtube channel"
                                        label="Destination Target Name"
                                        name="name"
                                        onChange={[Function]}
                                        type="text"
                                        value="" />
                              <DropDownMenu
                                        anchorOrigin={
                                                  Object {
                                                            "horizontal": "left",
                                                            "vertical": "top",
                                                          }
                                        }
                                        animated={true}
                                        autoWidth={false}
                                        disabled={false}
                                        iconButton={<NavigationArrowDropDown />}
                                        maxHeight={500}
                                        multiple={false}
                                        name="privacyStatus"
                                        onChange={[Function]}
                                        openImmediately={false}
                                        selectedMenuItemStyle={
                                                  Object {
                                                            "color": "#6908A1",
                                                          }
                                        }
                                        style={
                                                  Object {
                                                            "width": 310,
                                                          }
                                        }
                                        value="public">
                                        <MenuItem
                                                  anchorOrigin={
                                                            Object {
                                                                      "horizontal": "right",
                                                                      "vertical": "top",
                                                                    }
                                                  }
                                                  checked={false}
                                                  desktop={false}
                                                  disabled={false}
                                                  focusState="none"
                                                  insetChildren={false}
                                                  primaryText="Public"
                                                  value="public" />
                                        <MenuItem
                                                  anchorOrigin={
                                                            Object {
                                                                      "horizontal": "right",
                                                                      "vertical": "top",
                                                                    }
                                                  }
                                                  checked={false}
                                                  desktop={false}
                                                  disabled={false}
                                                  focusState="none"
                                                  insetChildren={false}
                                                  primaryText="Private"
                                                  value="private" />
                                        <MenuItem
                                                  anchorOrigin={
                                                            Object {
                                                                      "horizontal": "right",
                                                                      "vertical": "top",
                                                                    }
                                                  }
                                                  checked={false}
                                                  desktop={false}
                                                  disabled={false}
                                                  focusState="none"
                                                  insetChildren={false}
                                                  primaryText="Unlisted"
                                                  value="unlisted" />
                              </DropDownMenu>
                              <LoaderButton
                                        disabled={true}
                                        isLoading={undefined}
                                        loadingText="Creating target..."
                                        text="Create"
                                        type="submit" />
                    </div>
          </form>
</Modal>,
        "_debugID": 2,
        "_renderedOutput": <Modal
          closeModal={undefined}
          open={undefined}
          title="Create Youtube Target">
          <form
                    onSubmit={[Function]}>
                    <div
                              className={undefined}>
                              <LivepinTextField
                                        hint="My youtube channel"
                                        label="Destination Target Name"
                                        name="name"
                                        onChange={[Function]}
                                        type="text"
                                        value="" />
                              <DropDownMenu
                                        anchorOrigin={
                                                  Object {
                                                            "horizontal": "left",
                                                            "vertical": "top",
                                                          }
                                        }
                                        animated={true}
                                        autoWidth={false}
                                        disabled={false}
                                        iconButton={<NavigationArrowDropDown />}
                                        maxHeight={500}
                                        multiple={false}
                                        name="privacyStatus"
                                        onChange={[Function]}
                                        openImmediately={false}
                                        selectedMenuItemStyle={
                                                  Object {
                                                            "color": "#6908A1",
                                                          }
                                        }
                                        style={
                                                  Object {
                                                            "width": 310,
                                                          }
                                        }
                                        value="public">
                                        <MenuItem
                                                  anchorOrigin={
                                                            Object {
                                                                      "horizontal": "right",
                                                                      "vertical": "top",
                                                                    }
                                                  }
                                                  checked={false}
                                                  desktop={false}
                                                  disabled={false}
                                                  focusState="none"
                                                  insetChildren={false}
                                                  primaryText="Public"
                                                  value="public" />
                                        <MenuItem
                                                  anchorOrigin={
                                                            Object {
                                                                      "horizontal": "right",
                                                                      "vertical": "top",
                                                                    }
                                                  }
                                                  checked={false}
                                                  desktop={false}
                                                  disabled={false}
                                                  focusState="none"
                                                  insetChildren={false}
                                                  primaryText="Private"
                                                  value="private" />
                                        <MenuItem
                                                  anchorOrigin={
                                                            Object {
                                                                      "horizontal": "right",
                                                                      "vertical": "top",
                                                                    }
                                                  }
                                                  checked={false}
                                                  desktop={false}
                                                  disabled={false}
                                                  focusState="none"
                                                  insetChildren={false}
                                                  primaryText="Unlisted"
                                                  value="unlisted" />
                              </DropDownMenu>
                              <LoaderButton
                                        disabled={true}
                                        isLoading={undefined}
                                        loadingText="Creating target..."
                                        text="Create"
                                        type="submit" />
                    </div>
          </form>
</Modal>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <YoutubeTarget />,
}
`;
